dynamic_kick:
  ros__parameters:
    ### Engine ###

    # Distances and positions
    engine_rate: 500
    foot_rise: 0.08
    foot_distance: 0.2
    kick_windup_distance: 0.29
    trunk_height: 0.39
    trunk_roll: -0.051229096615671
    trunk_pitch: -0.136135681655558
    trunk_yaw: 0.0558505360638186

    # Timings
    move_trunk_time: 0.56
    raise_foot_time: 0.15
    move_to_ball_time: 0.19
    kick_time: 0.15
    move_back_time: 0.07
    lower_foot_time: 0.06
    move_trunk_back_time: 0.08

    # Decisions
    choose_foot_corridor_width: 0.4

    ### Stabilizer ###
    use_center_of_pressure: false
    stabilizing_point_x: -0.02
    stabilizing_point_y: -0.02

    # these values are used only if use_center_of_pressure is true
    pid_x/p: 0.2
    pid_x/i: 0
    pid_x/d: 0
    pid_y/p: 0.5
    pid_y/i: 0.0015
    pid_y/d: 1.5

    ### Visualization ###
    spline_smoothness: 100

    unstable:
      foot_distance: 0.21
      foot_rise: 0.11
      kick_speed: 6.7
      kick_time: 0.11
      kick_windup_distance: 0.33
      lower_foot_time: 0.16
      move_back_time: 0.26
      move_to_ball_time: 0.36
      move_trunk_back_time: 0.11
      move_trunk_time: 0.26
      raise_foot_time: 0.29
      stabilizing_point_x: 0.01
      stabilizing_point_y: -0.04
      trunk_height: 0.41
      trunk_pitch: -0.488692190558412
      trunk_roll: 0.102974425867665
      trunk_yaw: 0.296705972839036
      use_center_of_pressure: false
